internal/btf/btf.go

```diff
@@ -439,50 +439,29 @@ func (s *Spec) Program(name string, length uint64) (*Program, error) {
 	return &Program{s, length, funcInfos, lineInfos}, nil
 }
 
-// Map finds the BTF for a map.
+// Map returns the Struct holding a named map's members (fields), which can
+// be further processed into an eBPF map definition.
 //
-// Returns an error if there is no BTF for the given name.
-func (s *Spec) Map(name string) (*Map, []Member, error) {
-	var mapVar Var
-	if err := s.FindType(name, &mapVar); err != nil {
-		return nil, nil, err
+// This is a convenience wrapper that performs a lookup of a named Var
+// that is expected to contain a Struct. These Vars are implicitly created
+// by declaring a struct with a 'structure variable', as is common in eBPF C.
+// For example, `struct { ... } map_name ...` emits a global variable `map_name`
+// with the type of said struct (which can be of anonymous type).
+//
+// Returns an error if there is no BTF Var with the given name, or if the
+// Var is not a struct.
+func (s *Spec) Map(name string) (*Struct, error) {
```

Comment:
You're right that returning Struct and Var here is a bit weird. Do we add an accessor function for every `Type` in existence? I'd rather not.

Why not kill this whole function + `VarByName` and put the code into `loadBTFMaps`? The way it looks your follow up work on tail calls will make this obsolete anyways.

Reply:
P.S. `Spec.Datasec` is also a candidate for a function that is so short it doesn't do anything useful.

Reply:
Done! Let's revisit if/when we make BTF types public and the style of necessary accessors and convenience methods is more clear.